core/tests/protractor_desktop/profileFeatures.js

```diff
@@ -15,3 +15,105 @@
 /**
 * @fileoverview End-to-end tests for user profile features.
 */
+
+var general = require('../protractor_utils/general.js');
+var users = require('../protractor_utils/users.js');
+var workflow = require('../protractor_utils/workflow.js');
+
+
+var LearnerDashboardPage =
+  require('../protractor_utils/LearnerDashboardPage.js');
+var LibraryPage = require('../protractor_utils/LibraryPage.js');
+var SubscriptionDashboardPage =
+  require('../protractor_utils/SubscriptionDashboardPage.js');
+
+describe('Profile Features functionality', function() {
+  var libraryPage = null;
+  var learnerDashboardPage = null;
+  var subscriptionDashboardPage = null;
+
+  var creator1Id = 'FirstCreator';
+  var creator2Id = 'SecondCreator';
+  users.createUser('FirstUser@profileFeatures.com',
+    'FirstUserprofileFeatures');
+  users.createUser('SecondUser@profileFeatures.com',
+    'SecondUserprofileFeatures');
+  users.createUser(creator1Id + '@profileFeatures.com',
+    creator1Id);
+  users.createUser(creator2Id + '@profileFeatures.com',
+    creator2Id);
+
+  beforeAll(function() {
+    libraryPage = new LibraryPage.LibraryPage();
+    learnerDashboardPage = new LearnerDashboardPage.LearnerDashboardPage();
+    subscriptionDashboardPage =
+      new SubscriptionDashboardPage.SubscriptionDashboardPage();
+  });
+
+  it('View a particular exploration created by another User', function() {
```

Comment:
I don't understand this test. So first you subscribe to the creator, then I don't understand why you search from the library. I think you should rather start playing the exploration from the subscriptions page itself (or something like that, basically, there is no point of searching for the exploration from the library in this test).

Reply:
Because once the user subscribes, he should be able to see all the creations of a creator in his own library. So rather than playing exploration there itself, I'm searching in the user's library.

Reply:
Actually no. I think the flow that should happen is:
- subscribe to a user
- Go to the learner dashboard
- Go to the subscriptions tab
- Check that the user is there, and click on it
- You will be taken to the user's profile, where you can see all published explorations. You should play the exploration from here. 

Playing from the library doesn't fit into this user journey I think.

Reply:
In this case as well, there is no function defined to redirect to user's page from the learner's subscription tab.